internal/pkg/gateway/commit/notifier_test.go

```diff
@@ -193,5 +193,50 @@ func TestNotifier(t *testing.T) {
 
 			require.False(t, ok, "Expected notification channel to be closed but receive was successful")
 		})
+
+		t.Run("multiple listeners receive notifications", func(t *testing.T) {
+			commitSend := make(chan *ledger.CommitNotification, 2)
+			notifier := newTestNotifier(commitSend)
+
+			commitReceive1, _ := notifier.Notify(context.Background().Done(), "CHANNEL_NAME", "TX_ID")
+			commitReceive2, _ := notifier.Notify(context.Background().Done(), "CHANNEL_NAME", "TX_ID")
```

Comment:
Please add assertions on the errors.

```
func (notifier *Notifier) Notify(done <-chan struct{}, channelName string, transactionID string) (<-chan Notification, error) {
```

Reply:
I personally am not a fan of single character variables names other than specific, very well established (and generally cross-language) conventions such as `i` as a loop index.

If you are mandating that code must use single-character receiver names to be allowed into Fabric, please can you come out and say that so I can understand clearly, and create linting rules to require this.

Reply:
The "returns notifier on successful registration" test already explicitly tests the error value. What is the value of duplicating the same test at least once in every other test case?

Reply:
It's not a _test_, it's an _assertion_. It's a piece of the test to ensure that the expected behavior is maintained throughout the execution of the test.

Ignoring errors is *never* the right thing to do - even in tests. This has been a historical problem in the code base that I have spent *significant time* addressing over the past years. It's an established convention and code that comes into Fabric should honor it.

We've now spent as much time debating it as it would take to add `require.NoError(t, err)` at the call sites.

Reply:
Ok, I’ll remove the explicit test for this condition and add assertions to all the other test cases